src/androidTest/java/com/couchbase/lite/replicator/PullerInternalTest.java

```diff
@@ -0,0 +1,572 @@
+/**
+ * Copyright (c) 2015 Couchbase, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.couchbase.lite.replicator;
+
+import com.couchbase.lite.Document;
+import com.couchbase.lite.LiteTestCaseWithDB;
+import com.couchbase.lite.mockserver.MockChangesFeed;
+import com.couchbase.lite.mockserver.MockChangesFeedNoResponse;
+import com.couchbase.lite.mockserver.MockCheckpointPut;
+import com.couchbase.lite.mockserver.MockDispatcher;
+import com.couchbase.lite.mockserver.MockDocumentBulkGet;
+import com.couchbase.lite.mockserver.MockDocumentGet;
+import com.couchbase.lite.mockserver.MockHelper;
+import com.couchbase.lite.util.Log;
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by hideki on 12/15/15.
+ */
+public class PullerInternalTest  extends LiteTestCaseWithDB {
```

Comment:
Would it be better to be ReplicatorInternal (iOS has ReplicatorInternal_Tests) so it can be used for replicator internal in general?